src/components/Post/Post.jsx

```diff
@@ -4,52 +4,337 @@ import {
   IconButton,
   Avatar,
   Button,
+  TextField,
   Box,
 } from "@mui/material";
+import ListItem from "@mui/material/ListItem";
+import Divider from "@mui/material/Divider";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import Modal from "@mui/material/Modal";
+
 import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
 import CommentIcon from "@mui/icons-material/Comment";
 import BookmarkBorderIcon from "@mui/icons-material/BookmarkBorder";
 import CardMedia from "@mui/material/CardMedia";
+import FavoriteIcon from "@mui/icons-material/Favorite";
+import BookmarkIcon from "@mui/icons-material/Bookmark";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import {
+  updateLikedPost,
+  updateCommentInPost,
+  deleteCommentInPost,
+  updatePostsForDelete,
+} from "features/post/postSlice";
+import {
+  requestLikePost,
+  requestUnlikePost,
+  requestPostBookmark,
+  requestRemovePostFromBookmark,
+  requestAddComment,
+  requestDeleteComment,
+  requestDeletePost,
+} from "../../firebase/firestore-requests";
+import { useSelector, useDispatch } from "react-redux";
+
+import {
+  updateUserLikedPost,
+  updateUserUnlikedPost,
+  updateUserBookmarkedPost,
+  updateUserRemoveBookmarkedPost,
+  deleteUserPost,
+} from "features/user/userSlice";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+
+import { EditPost } from "components/EditPost/EditPost";
 
 export const Post = ({ post }) => {
-  console.log(post);
+  const [isCommentsActive, setIsCommentsActive] = useState(false);
+  const [commentText, setCommentText] = useState("");
+
+  const {
+    userDetails: { likedPost, bookmarks },
+  } = useSelector((store) => store.userDetails);
+  const { token } = useSelector((store) => store.authDetails);
+  const { users } = useSelector((store) => store.allUsers);
+
+  const dispatch = useDispatch();
+
+  const postBy = users.find((user) => user.id === post?.data.postBy);
+  const isLiked = likedPost?.likedPost?.some((_id) => _id === post?.id);
+  const isBookmarked = bookmarks?.bookmarks?.some((_id) => _id === post?.id);
+
+  const handleLike = async (e) => {
+    try {
+      await requestLikePost(post.id, token, postBy.id);
+      dispatch(updateLikedPost({ id: post.id, count: 1 }));
+      dispatch(updateUserLikedPost(post.id));
+    } catch (error) {
+      console.log(error);
+    }
+  };
+  const handleUnLike = async (e) => {
+    try {
+      await requestUnlikePost(post.id, token, postBy.id);
+      dispatch(updateLikedPost({ id: post.id, count: -1 }));
+      dispatch(updateUserUnlikedPost(post.id));
+    } catch (error) {
+      console.log(error);
+    }
+  };
```

Comment:
These post action calls can be moved to separate files

Reply:
Thanks for the suggestion, will do it later